python-package/lightgbm/callback.py

```diff
@@ -231,124 +231,135 @@ def early_stopping(stopping_rounds: int, first_metric_only: bool = False, verbos
+                    self.best_score.append(float('inf'))
+                    self.cmp_op.append(partial(_lt_delta, delta=delta))
+
+        def _final_iteration_check(self, env: CallbackEnv, eval_name_splitted: List[str], i: int) -> None:
+            if env.iteration == env.end_iteration - 1:
+                if self.verbose:
+                    best_score_str = '\t'.join(
+                        [_format_eval_result(x) for x in self.best_score_list[i]])
+                    _log_info('Did not meet early stopping. '
+                              f'Best iteration is:\n[{self.best_iter[i] + 1}]\t{best_score_str}')
+                    if self.first_metric_only:
                         _log_info(f"Evaluated only: {eval_name_splitted[-1]}")
-                raise EarlyStopException(best_iter[i], best_score_list[i])
-            _final_iteration_check(env, eval_name_splitted, i)
-    _callback.order = 30  # type: ignore
-    return _callback
+                raise EarlyStopException(
+                    self.best_iter[i], self.best_score_list[i])
+
+        def __call__(self, env: CallbackEnv) -> None:
+            if env.iteration == env.begin_iteration:
+                self._init(env)
+            if not self.enabled:
+                return
+            for i in range(len(env.evaluation_result_list)):
+                score = env.evaluation_result_list[i][2]
+                if self.best_score_list[i] is None or self.cmp_op[i](score, self.best_score[i]):
+                    self.best_score[i] = score
+                    self.best_iter[i] = env.iteration
+                    self.best_score_list[i] = env.evaluation_result_list
+                # split is needed for "<dataset type> <metric>" case (e.g. "train l1")
+                eval_name_splitted = env.evaluation_result_list[i][1].split(
+                    " ")
+                if self.first_metric_only and self.first_metric != eval_name_splitted[-1]:
+                    continue  # use only the first metric for early stopping
+                if ((env.evaluation_result_list[i][0] == "cv_agg" and eval_name_splitted[0] == "train"
+                        or env.evaluation_result_list[i][0] == env.model._train_data_name)):
+                    self._final_iteration_check(env, eval_name_splitted, i)
+                    # train data for lgb.cv or sklearn wrapper (underlying lgb.train)
+                    continue
+                elif env.iteration - self.best_iter[i] >= self.stopping_rounds:
+                    if self.verbose:
+                        eval_result_str = '\t'.join(
+                            [_format_eval_result(x) for x in self.best_score_list[i]])
+                        _log_info(
+                            f"Early stopping, best iteration is:\n[{self.best_iter[i] + 1}]\t{eval_result_str}")
+                        if self.first_metric_only:
+                            _log_info(
+                                f"Evaluated only: {eval_name_splitted[-1]}")
```

Comment:
Could you please revert indentation changes like these? That would simplify the diff and make it easier for reviewers to understand what is being changed.

For example, on `master` this project has

```python
eval_result_str = '\t'.join([_format_eval_result(x) for x in best_score_list[i]])
```

and on this PR you have

```python
eval_result_str = '\t'.join(
                            [_format_eval_result(x) for x in self.best_score_list[i]])
```

We don't enforce a limit on line length in this project's Python code, and prefer to leave style-only changes to maintainers in most cases (to ensure that the git blame is easy to use as a source of information about when lines changed).

Reply:
Hey @jameslamb, I couldn't find any information on what formatting style to use so I just ran autopep8. Let me try to make the formatting as close to original as possible.